engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -81,7 +85,11 @@ class NextgenEngine @JvmOverloads constructor(nadel: Nadel, transforms: List<Nad
             fieldToService.getServicesForTopLevelFields(query)
                 .map { (field, service) ->
                     async {
-                        executeTopLevelField(field, service, executionInput)
+                        val serviceContext = serviceExecutionHooks.createServiceContext(
+                            CreateServiceContextParams.newParameters().service(service).build()
+                        )
+
+                        executeTopLevelField(field, service.withContext(serviceContext.await()), executionInput)
```

Comment:
I think you need to create a new `ServiceContext` for the `executeHydration` function too.

I wish `executeHydration` didn't exist, but without a combined schema, we need it 😞.

Reply:
Hm, I think we won't need to do that if put the context into `NadelExecutionContext` (so many contexts 😄). 
`NadelExecutionContext` is passed around to hydrations too, so they can use the same instance of `ServiceContext`.

Also, having more than 1 `ServiceContext` instance per `Service` kind of defeats the purpose of having a `ServiceContext`.